### PR TITLE
Upgrade Mongoid to version 9.0.0

### PR DESCRIPTION
**Pull request by sairamsrinivasan**
This PR upgrades mongoid to version 9.0.0

### CODE REVIEW

File: spec/models/profile.rb [hunk 1]
```diff
@@ -6,7 +6,7 @@ class Profile
   field :age
   field :hobbies, type: Array, default: []
 
-  embedded_in :user, inverse_of: :profile
+  embedded_in :user, inverse_of: :profile, touch: false
```
Comment (SairamSrinivasan) [inline spec/models/profile.rb:9]: Embedded associations now have the touch option defaulted to `true`: https://www.mongodb.com/docs/mongoid/master/release-notes/mongoid-9.0/#embedded_in-associations-now-default-to-touch--true

**Comment by SairamSrinivasan** [comment]
Ready for review.

**Review by dblock — CHANGES_REQUESTED** [review]
Add to CI please?

https://github.com/mongoid/mongoid-rspec/blob/master/.github/workflows/rspec.yml#L17

**Comment by sairamsrinivasan** [comment]
@dblock - I appreciate the review. Please check out the latest revisions.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,7 +1,8 @@
-### 4.1.1 (Next)
+### 4.2.0 (Next)
 
 * [#239](https://github.com/mongoid/mongoid-rspec/pull/239): Add support for if/unless validator options - [@knovoselic](https://github.com/knovoselic).
-* [#244](https://github.com/mongoid/mongoid-rspec/pull/123): Migration from TravisCI to GitHub Actions - [@skalibog](https://github.com/skalibog).
+* [#244](https://github.com/mongoid/mongoid-rspec/pull/244): Migration from TravisCI to GitHub Actions - [@skalibog](https://github.com/skalibog).
+* [#245](https://github.com/mongoid/mongoid-rspec/pull/245): Added Support for Mongoid v9.0.0 - [@saisrinivasan](https://github.com/SairamSrinivasan).
```
Comment (dblock) [inline CHANGELOG.md:5]: Let's match the rest of CHANGELOG: "Add support for Mongoid 9"  (Add vs. Added, lowercase support, generalize 9).

**Review by dblock — CHANGES_REQUESTED** [review]
One more nit, apologies for my OCD, looking good.

**Comment by sairamsrinivasan** [comment]
> One more nit, apologies for my OCD, looking good.

No worries. Thanks for being thorough. I pushed an update.

This is unrelated but any chance that you or the other maintainers can look at my other PR: https://github.com/mongoid/mongoid-rspec/pull/246? I will likely need to rebase this.

**Comment by dblock** [comment]
> This is unrelated but any chance that you or the other maintainers can look at my other PR: #246? I will likely need to rebase this.

Yes, I was doing this one first. Rebase and I'll take a look.

**Pull request merged**